UnitTests/Mono.Debugging.Tests/Shared/DebugTests.cs

```diff
@@ -256,12 +256,18 @@ void GetLineAndColumn (string breakpointMarker, int offset, string statement, ou
 		}
 
 		public Breakpoint AddBreakpoint (string breakpointMarker, int offset = 0, string statement = null, ITextFile file = null)
+		{
+			var bp = CreateBreakpoint (breakpointMarker, offset, statement, file);
+			Session.Breakpoints.Add (bp);
+			return bp;
+		}
+
+		public Breakpoint CreateBreakpoint (string breakpointMarker, int offset = 0, string statement = null, ITextFile file = null)
 		{
 			file = file ?? SourceFile;
 			int col, line;
 			GetLineAndColumn (breakpointMarker, offset, statement, out line, out col, file);
 			var bp = new Breakpoint (file.Name, line, col);
-			Session.Breakpoints.Add (bp);
```

Comment:
This is so weird, isn't this all synchronous?

Reply:
Ohhh, I get it now.

Reply:
The Breakpoint class has a CommitChanges method which would probably be another way to do this, the breakpoints dialog does this when updating an existing breakpoint, but that would likely trigger two updates in the vscode debugger session for the same breakpoint - seemed easier to have everything ready when the breakpoint is added.